Implemented CandidateValidator and solved tasks

DanaKhromenko opened the pull request.

DanaKhromenko commented:
- removed extra separations in import groups;
- changed order for imports;
- separated line that was longer than 100 characters;
- replaced '.*' import with concrete imports.

RomanDubovskyi commented on src/main/java/practice/CandidateValidator.java:
```diff
@@ -1,5 +1,27 @@
+        if (periodsInUkr == null || periodsInUkr.length() != 9) {
+            return 0;
+        }
+        try {
```
тут краще щоб випала помилка, бо якщо ми не змогли розпарсити вхідні параметри цей кейс не потрібно опрацьовувати. Дані можуть бути валідними просто з опечаткою, в такому випадку валідатор не має казати, що кандидат непридатний

RomanDubovskyi commented on src/main/java/practice/StreamPractice.java:
```diff
@@ -14,7 +20,13 @@
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return numbers.stream()
+                .map(s -> s.split(","))
+                .flatMap(Arrays::stream)
```
```suggestion
                .flatMap(n -> Arrays.stream(n.split(",")))
```

RomanDubovskyi commented on src/main/java/practice/StreamPractice.java:
```diff
@@ -14,7 +20,13 @@
+                .mapToInt(Integer::parseInt)
+                .filter(i -> i % 2 == 0)
+                .min()
+                .orElseThrow(() -> new RuntimeException("Can't get min value from list"));
```
https://mate-academy.github.io/jv-program-common-mistakes/java-core/java-eight-part-two/stream-api

DanaKhromenko replied:
На что именно ты здесь указываешь?

DanaKhromenko replied:
Если ты про то, что в ошибку надо добавить " + numbers", то done

RomanDubovskyi commented on src/main/java/practice/StreamPractice.java:
```diff
@@ -23,7 +35,19 @@ public int findMinEvenNumber(List<String> numbers) {
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        int subtractValue = 1;
+        Stream<Integer> oddStream = IntStream.range(0, numbers.size())
```
```suggestion
       return IntStream.range(0, numbers.size())
          .map(i -> i % 2 != 0 ? (numbers.get(i) - 1) : numbers.get(i))
          .filter(n -> n % 2 != 0)
          .... (avg + get)
```

DanaKhromenko replied:
Ой, как круто!!!

RomanDubovskyi commented on src/main/java/practice/StreamPractice.java:
```diff
@@ -35,7 +59,11 @@ public Double getOddNumsAverage(List<Integer> numbers) {
     public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> p.getSex() == Person.Sex.MAN)
```
single filter operation will work faster than 3, use single condition with logical operators

DanaKhromenko commented:
- removed checks in yearInUkraine-method;
- replaced map+flatmap with flatmap in findMinEvenNumber-method;
- changed exception message in findMinEvenNumber-method;
- replaced 3 streams with 1 in getOddNumsAverage-method;
- replaced 3 filters with 1 predicate in selectMenByAge-method.

RomanDubovskyi reviewed (approved):
GJ👍
